src/Selenium2Driver.php

```diff
@@ -1120,4 +1132,52 @@ private function trigger($xpath, $event, $options = '{}')
         $script = 'Syn.trigger("' . $event . '", ' . $options . ', {{ELEMENT}})';
         $this->withSyn()->executeJsOnXpath($xpath, $script);
     }
+
+    /**
+     * Uploads a file to the Selenium instance.
+     *
+     * Note that uploading files is not part of the official WebDriver
+     * specification, but it is supported by Selenium.
+     * @see https://github.com/SeleniumHQ/selenium/blob/master/py/selenium/webdriver/remote/webelement.py#L533
+     *
+     * @param string $path     The path to the file to upload.
+     *
+     * @return string          The remote path.
+     *
+     * @throws DriverException When the file is not found.
+     * @throws InvalidRequest  When the driver does not support file uploads.
+     */
+    public function uploadFile($path) {
+        if (!is_file($path)) {
+          throw new DriverException('Could not upload file, the file: ' . $path . '. was not found.');
+        }
+
+        // Selenium only accepts uploads that are compressed as a Zip archive.
+        $temp_filename = tempnam('', 'WebDriverZip');
+
+        $archive = new \ZipArchive();
```

Comment:
Also it seems, that you always are using `ZipArchive` class now.

To preserve BC I think, that:

* in case if class is absent and we know it's remote file (not exists locally) then throw `DriverException`
* in case case if class is absent and it's local file just upload file the old way without even creating `ZipArchive`